Fix discard_stale (Closes #65)

g-gemignani opened the pull request:
Make sure that analyzers flagged with discard_stale = true are correctly
removed after being stale for a period greater than the timeout

g-gemignani committed:
Address issue 65

Make sure that analyzers flagged with discard_stale = true are correctly
removed after being stale for a period greater than the timeout

g-gemignani committed:
Fix copyright and remove unused imports

trainman419 commented on diagnostic_aggregator/test/discard_stale_not_published_test.py:
```diff
@@ -0,0 +1,80 @@
+            for stat in msg.status:
+                self._agg_expecteds.append(stat)
+
+    def test_discard_stale(self):
```
In order to properly test the original bug, this test should publish some diagnostic items with the `nonexistent2` prefix, and then wait for them to go stale and be removed before checking that they're no longer published by the diagnostic aggregator.

g-gemignani replied:
addressed the issue

g-gemignani committed:
Address pr issue about discard_stale test

g-gemignani committed:
Fix problem with pr that skipped the timeout stale transition period

g-gemignani commented:
Hello,
thank you for the feedback on the pr. Indeed I found a problem with it, where actually the timeout period of stale time was skipped. I addressed the issue and updated the tests.
Thank you!

trainman419 commented:
Thanks!
